### PR TITLE
update dotnetcore path for vso image and install xdebug for php and some  ruby tools as well

### PR DESCRIPTION
**Pull request by arroyc**
- [X] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [X] Tests are included and/or updated for code changes.
- [X] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/PlatformInstallerBase.cs [hunk 1]
```diff
@@ -71,10 +72,20 @@ public abstract class PlatformInstallerBase
                 .AppendLine($"echo \"$checksumValue {version}.tar.gz\" | sha512sum -c - >/dev/null 2>&1")
                 .AppendLine("echo Extracting contents...")
                 .AppendLine($"tar -xzf {tarFile} -C .")
-                .AppendLine($"rm -f {tarFile}")
+//                .AppendLine($"rm -f {tarFile}")
```
Comment (qianz2) [inline src/BuildScriptGenerator/PlatformInstallerBase.cs:75]: clean up tar file still needed for platforms other than dotnet?

File: images/build/Dockerfiles/vso.focal.Dockerfile [hunk 2]
```diff
@@ -121,7 +121,7 @@ RUN set -ex \
     && . $buildDir/__dotNetCoreSdkVersions.sh \
     && ln -s $DOT_NET_CORE_21_SDK_VERSION 2-lts \
     && ln -s $DOT_NET_CORE_31_SDK_VERSION 3-lts \
-    && ln -s 3-lts lts \
+    && ln -s 3-lts lts \
```
Comment (qianz2) [inline images/build/Dockerfiles/vso.focal.Dockerfile:124]: nit: space

File: src/BuildScriptGenerator/PlatformInstallerBase.cs [hunk 3]
```diff
@@ -42,6 +42,7 @@ public abstract class PlatformInstallerBase
             }
 
             var tarFile = $"{version}.tar.gz";
+
```
Comment (qianz2) [inline src/BuildScriptGenerator/PlatformInstallerBase.cs:45]: nit: space

File: src/BuildScriptGenerator/PlatformInstallerBase.cs [hunk 4]
```diff
@@ -71,10 +72,16 @@ public abstract class PlatformInstallerBase
                 .AppendLine($"echo \"$checksumValue {version}.tar.gz\" | sha512sum -c - >/dev/null 2>&1")
                 .AppendLine("echo Extracting contents...")
                 .AppendLine($"tar -xzf {tarFile} -C .")
-                .AppendLine($"rm -f {tarFile}")
+                .AppendLine($"rm -f {tarFile}")
```
Comment (qianz2) [inline src/BuildScriptGenerator/PlatformInstallerBase.cs:75]: nit: space

File: build/vsoSymlinksDotNetCore.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+allSdksDir="/home/codespace/.dotnet"
+mkdir -p "$allSdksDir"
+
+# Copy latest muxer and license files
+cp -f "$splitSdksDir/lts/dotnet" "$allSdksDir"
+cp -f "$splitSdksDir/lts/LICENSE.txt" "$allSdksDir"
+cp -f "$splitSdksDir/lts/ThirdPartyNotices.txt" "$allSdksDir"
+
+function createLinks() {
```
Comment (qianz2) [inline build/vsoSymlinksDotNetCore.sh:19]: why do we want to create links, what'll be the purpose for using it

File: src/BuildScriptGenerator/PlatformInstallerBase.cs [hunk 6]
```diff
@@ -71,10 +72,16 @@ public abstract class PlatformInstallerBase
...
-                .AppendLine($"rm -f {tarFile}")
+                .AppendLine($"rm -f {tarFile}")             
                 .AppendLine("PLATFORM_SETUP_ELAPSED_TIME=$(($SECONDS - $PLATFORM_SETUP_START))")
                 .AppendLine("echo \"Done in $PLATFORM_SETUP_ELAPSED_TIME sec(s).\"")
                 .AppendLine("echo")
+                .AppendLine("oryxImageDetectorFile=\"/opt/oryx/.imagetype\"")
+                .AppendLine($"platformName=\"{platformName}\"")
+                .AppendLine($"if [ -f \"$oryxImageDetectorFile\" ] && [ \"$platformName\" = \"dotnet\" ] && grep -q \"vso-focal\" \"$oryxImageDetectorFile\"; then")
+                .AppendLine("echo \"image detector file exists, platform is dotnet..\"")
+                .AppendLine($"source /opt/tmp/build/vsoSymlinksDotNetCore.sh")
```
Comment (qianz2) [inline src/BuildScriptGenerator/PlatformInstallerBase.cs:83]: Do we need these lines in this file? We've added `RUN ./opt/tmp/build/vsoSymlinksDotNetCore.sh` in `vso.focal.dockerfile`  so we don't have to source the script at here?

File: build/vsoSymlinksDotNetCore.sh (hunk 5 above)
Comment (arroyc) [reply build/vsoSymlinksDotNetCore.sh:19]: vso/codespaces team need this links to be created as per latest guidelines from the dotnetcore team to keep multiple dotnet sdk... so we need these links to standardize it

File: images/build/Dockerfiles/vso.focal.Dockerfile [hunk 7]
```diff
@@ -240,11 +240,22 @@ RUN buildDir="/opt/tmp/build" \
...
     && echo "vso-focal" > /opt/oryx/.imagetype
 
+# install few more tools for VSO
+RUN gem install bundler rake ruby-debug-ide debase
+RUN  yes | pecl install xdebug \
+    && export PHP_LOCATION=$(dirname $(dirname $(which php))) \
+    && echo "zend_extension=$(find ${PHP_LOCATION}/lib/php/extensions/ -name xdebug.so)" > ${PHP_LOCATION}/ini/conf.d/xdebug.ini \
+    && echo "xdebug.mode = debug" >> ${PHP_LOCATION}/ini/conf.d/xdebug.ini \
+    && echo "xdebug.start_with_request = yes" >> ${PHP_LOCATION}/ini/conf.d/xdebug.ini \
+    && echo "xdebug.client_port = 9000" >> ${PHP_LOCATION}/ini/conf.d/xdebug.ini
```
[the remark this answers is not in the document]
Comment (arroyc) [reply images/build/Dockerfiles/vso.focal.Dockerfile:254]: yes ... this is fixed port for xdebug .. i believe this is the port vscode xdebug extension gets connected

File: src/BuildScriptGenerator/PlatformInstallerBase.cs (hunk 6 above)
Comment (arroyc) [reply src/BuildScriptGenerator/PlatformInstallerBase.cs:83]: no we need this script for dynamic installation scenario .. this script needs during a dynamic installation to keep the links up to date.

**Review by qianz2 — APPROVED** [review]

**Pull request merged**